packages/sanity/src/core/util/draftUtils.ts

```diff
@@ -65,17 +74,53 @@ export function getIdPair(id: string): {draftId: DraftId; publishedId: Published
 
 /** @internal */
 export function isPublishedId(id: string): id is PublishedId {
-  return !isDraftId(id)
+  return !isDraftId(id) && !isVersionId(id)
 }
 
 /** @internal */
 export function getDraftId(id: string): DraftId {
   return isDraftId(id) ? id : ((DRAFTS_PREFIX + id) as DraftId)
 }
 
+/**  @internal */
+export function getVersionId(id: string, bundle: string): string {
+  if (isVersionId(id)) {
+    const [_versionPrefix, versionId, ...publishedId] = id.split('.')
```

Comment:
Just a tiny thought: it might be nice to split on the `PATH_SEPARATOR` constant instead of `"."`.
```suggestion
    const [_versionPrefix, versionId, ...publishedId] = id.split(PATH_SEPARATOR)
```

Reply:
Absolutely, thanks!